src/System.Private.CoreLib/src/System/Diagnostics/Eventing/ManagedRuntimeEventSource.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+using System.Runtime.InteropServices;
+using Microsoft.Win32;
+using Microsoft.Win32.SafeHandles;
+
+using System.Runtime.InteropServices.ComTypes;
+using System.StubHelpers;
+
+
+namespace System.Diagnostics.Tracing
+{
+#if FEATURE_PERFTRACING
+    /// <summary>
+    /// ManagedRuntimeEventSource is an EventSource that represents events emitted by the managed runtime.
+    /// </summary>
+    [EventSource(Name = "System.Runtime")]
```

Comment:
Should the event source `Name` be in sync with the actual type name - do we need to have `Managed` in the type name? 

`Managed` is implied in the C# type names. We use the additional classifications (`Unmanaged`, `Native`, ...) for things that are not managed. If we expect to have non-managed runtime event source as well, we should make that one to have special suffix instead.

Reply:
I actually talked with @sywhang about this.   We happen to already use the name RuntimeEventSource to be the special managed façade for the native events generated by EventPipe.    I indeed recommended that the ideal solution would be to rename that one to be NativeRuntimeEventSource (or RuntimeEventSourceNative) and then use RuntimeEventSource for this managed one.    however there is some magic associated with that type, so it is not a trivial rename (but also not really that hard).    

@sywhang - are you willing to take a wack at the rename (do it as a separate PR)?

Reply:
The suffix ```Managed``` was put there because we already have the class ```RuntimeEventSource``` for the EventSource ```Microsoft-Windows-DotNETRuntime``` which contains events that are triggered from within the native runtime. @vancem and I talked about renaming the existing one to ```NativeRuntimeEventSource``` and changing this to ```RuntimeEventSource``` - should I do that instead of naming this ```ManagedRuntimeEventSource```?

Reply:
@vancem Yeah I'll put this PR on hold and submit a separate PR on renaming the existing one to ```NativeRuntimeEventSource```. 

Do you prefer ```NativeRuntimeEventSource``` or ```RuntimeEventSourceNative```?

Reply:
`NativeRuntimeEventSource`